//=================================================================================================
// project: das - data access service
// company: viga ltd.
// date   : 2016.12.17
//=================================================================================================
package das.proto;

//=================================================================================================
//PostHeartBeat
//=================================================================================================
message PostHeartBeat {
    required uint32 seqno        = 1;	//本包的包序号
    optional string id           = 2;	//身份ID
} 

//=================================================================================================
//AckHeartBeat
//=================================================================================================
message AckHeartBeat {
    required uint32 seqno        = 1;	//本包的包序号
}

//=================================================================================================
//SyncDeviceList
//=================================================================================================
message SyncFZUserList {
    required uint32 seqno           = 1;    //本包的包序号
}

//=================================================================================================
//AckSyncDeviceList
//=================================================================================================
message AckSyncFZUserList {
    required uint32 seqno           = 1;    //本包的包序号
    required int32  result          = 2;    //同步结果
    repeated string id              = 3;    //设备ID表
}

//=================================================================================================
//UpdateDeviceList
//=================================================================================================
message UpdateFZUserList {
    required uint32 seqno           = 1;	//本包的包序号
    required uint32 operation       = 2;	//删除(0)/添加(1)/禁用
    repeated string id              = 3;	//设备ID表
}

//=================================================================================================
//AckUpdateDeviceList
//=================================================================================================
message AckUpdateFZUserList {
    required uint32 seqno     	    = 1;    //本包的包序号
    required int32  result    	    = 2;    //更新结果
}

//=================================================================================================
//RequestNewGS
//=================================================================================================
message RequestNewFZUser {
    required uint32 seqno       = 1;    //本包的包序号
    required string user        = 2;    //用户的ID leng>=3
    optional string phone       = 3;    //验证手机
    optional string check       = 4;    //验证码
    optional string password    = 5;    //用户的密码
} 

//=================================================================================================
//AckNewGS
//=================================================================================================
message AckNewFZUser {
    required uint32 seqno   = 1;    //本包的包序号
    required int32  result  = 2;    //申请账户成功//-3:验证码错误,-2:名字不合法,-1:服务错误,0：名字存在
    optional string check   = 3;    //验证码 没有验证码存在，获取验证码
}

//=================================================================================================
//RequestFZUserIdentity
//=================================================================================================
message RequestFZUserIdentity {
    required uint32 seqno         = 1;      //本包的包序号
    required string user          = 2;      //用户
    required string pswd          = 3;      //密码
    required string pcsn          = 4;      //pc序列号
}

//=================================================================================================
//AckFZUserIdentity
//=================================================================================================
message AckFZUserIdentity {
    required uint32   seqno       = 1;    //本包的包序号
    required int32    result      = 2;    //ID分配的结果
    required int64    ver         = 3;    //软件注册版本<0未注册
    optional string   extradata   = 4;    //扩展备用数据
}

//=================================================================================================
//PostChangeFZPswd
//=================================================================================================
message PostChangeFZPswd {
    required uint32 seqno       = 1;    //本包的包序号
    required string old         = 2;    //原密码
    required string pswd        = 3;    //新密码
}

//=================================================================================================
//AckChangeFZPswd
//=================================================================================================
message AckChangeFZPswd {
    required uint32 seqno       = 1;    //本包的包序号
    required int32  result      = 2;    //修改结果
}

//=================================================================================================
//GSMsgType
//=================================================================================================
enum FZMsgType {
    DeleteFriend        = 1;  //删除好友
    RequestFriend       = 2;  //好友申请
    AgreeFriend         = 3;  //同意好友
    RejectFriend        = 4;  //拒绝好友
    Text                = 5;  //文字
    Land                = 6;  //地块
}

//=================================================================================================
//AckFZUserMessage仿真用户之间的消息
//=================================================================================================
message FZUserMessage {
    required uint32 seqno   = 1;    //本包的包序号
    required FZMsgType type = 2;    //消息类型
    required string from    = 3;    //消息发送者
    required string to      = 4;    //消息接收者
    optional bytes content  = 5;    //具体消息 
}

//=================================================================================================
//AckFZUserMessage
//=================================================================================================
message AckFZUserMessage {
    required uint32 seqno     = 1;  //本包的包序号
    required uint32 res       = 2;  //-1:不是好友; 0:不在线; 1:接收
    required string user      = 3;  //好友ID
}

//=================================================================================================
//RequestFriends地面站之间的消息
//=================================================================================================
message RequestFriends {
    required uint32 seqno       = 1;    //本包的包序号
}

//=================================================================================================
//RequestFriends地面站之间的消息
//=================================================================================================
message AckFriends {
    required uint32 seqno     = 1;  //本包的包序号
    repeated string friends   = 2;  //好友列表
}

//=================================================================================================
//UpdateSWKey
//=================================================================================================
message UpdateSWKey {
	required uint32 seqno         = 1;      //本包的包序号
	required string	swkey         = 2;      //软件序列号
	optional string	dscr          = 3;      //用户描述
	optional int64  ver           = 4;      //软件版本号
	optional int32  change        = 5;      //change:0新增，1修改，2解绑设备
}

//=================================================================================================
//AckUpdateSWKey
//=================================================================================================
message AckUpdateSWKey {
	required uint32 seqno         = 1;      //本包的包序号
	required int32 result         = 2;      //结果0:reject -1:无权限 -2:sn不合规范
	required string	swkey         = 3;      //软件序列号
}

//=================================================================================================
//ReqSWKeyInfo
//=================================================================================================
message ReqSWKeyInfo {
	required uint32 seqno         = 1;      //本包的包序号
	optional string	swkey         = 2;      //软件序列号
	optional string pcsn          = 3;      //pc序列号
	optional int64  begTm         = 4;      //生成时间
}

//=================================================================================================
//SWKeyInfo
//=================================================================================================
message SWKeyInfo {
	required string	swkey         = 1;      //软件序列号
	required int64  ver           = 2;      //软件版本号
	required int64  genTm         = 3;      //注册时间
	required int32  used          = 4;      //是否使用，0:没有
	optional string	dscr          = 5;      //用户描述
	optional string pcsn          = 6;      //pc序列号
	optional int64  regTm         = 7;      //注册时间
}

//=================================================================================================
//AckSWKeyInfo
//=================================================================================================
message AckSWKeyInfo {
	required uint32      seqno     = 1;      //本包的包序号
    repeated SWKeyInfo   keys      = 2;      //序列号
}

//=================================================================================================
//SWRegist
//=================================================================================================
message SWRegist {
    required uint32    seqno         = 1;      //本包的包序号
    required string    pcsn          = 2;      //主板sn
    required string    swkey         = 3;      //软件序列号
}

//=================================================================================================
//AckSWRegist
//=================================================================================================
message AckSWRegist {
    required uint32 seqno         = 1;      //本包的包序号
    required int32 result         = 2;      //结果0:reject
    required int64 ver            = 3;      //序列号版本
}

//=================================================================================================
//TestInfo
//=================================================================================================
message TestInfo {
    required string id        = 1;  //身份证号
    required string name      = 2;  //姓名
    required string sex       = 3;  //性别
    required string address   = 4;  //住址
    required string school    = 5;  //学校
    required string cag       = 6;  //班级
    required string stdid     = 7;  //学号
    required string examiner  = 8;  //考官
    required string comment   = 9;  //备注
}

//=================================================================================================
//FZResult
//=================================================================================================
message FZResult {
    optional int64 id         = 1;      //id
    optional int64 begTm      = 2;      //开始时间
    required int32 usedTm     = 3;      //用时
    required int32 type       = 4;      //类型
    required int32 rslt       = 5;      //成绩
    optional TestInfo info    = 6;      //考证信息
}

//=================================================================================================
//PostFZResult
//=================================================================================================
message PostFZResult {
    required uint32   seqno        = 1;      //本包的包序号
    required FZResult rslt         = 2;
}

//=================================================================================================
//AckPostFZResult
//=================================================================================================
message AckPostFZResult {
    required uint32   seqno        = 1;      //本包的包序号
    required int64 	  id           = 2;
}

//=================================================================================================
//RequestFZResult
//=================================================================================================
message RequestFZResults {
    required uint32 seqno        = 1;      //本包的包序号
    optional int64  tmbeg        = 2;      //考核起始时间
    optional int64  tmend        = 3;      //考核结束时间
}

//=================================================================================================
//RequestFZResult
//=================================================================================================
message AckFZResults {
    required uint32 seqno        = 1;      //本包的包序号
    repeated FZResult rslt       = 2;
}

//=================================================================================================
//FZInfo
//=================================================================================================
message FZInfo {
    optional string name        = 1; //姓名
    optional string grade       = 2; //年级
    optional string majr        = 3; //专业
    optional string id          = 4; //学号
    optional string school      = 5; //学校
    optional string births      = 6; //出生日期
    required string email       = 7; //e-mail
}

//=================================================================================================
//PostFZInfo
//=================================================================================================
message PostFZInfo {
    required uint32  seqno        = 1;      //本包的包序号
    required FZInfo  info         = 2;
}

//=================================================================================================
//AckPostFZInfo
//=================================================================================================
message AckPostFZInfo {
    required uint32  seqno     = 1;      //本包的包序号
    required int32   rslt      = 2;      //0:1
}

//=================================================================================================
//RequestFZInfo
//=================================================================================================
message RequestFZInfo {
    required uint32 seqno        = 1;      //本包的包序号
}

//=================================================================================================
//AckFZInfo
//=================================================================================================
message AckFZInfo {
    required uint32  seqno     = 1;      //本包的包序号
    optional FZInfo  info      = 2;      //
}

//=================================================================================================
//PostGetFZPswd
//=================================================================================================
message PostGetFZPswd {
    required uint32  seqno     = 1;      //本包的包序号
    required string  email     = 2;      //
    required string  user      = 3;      //
}

//=================================================================================================
//AckGetFZPswd
//=================================================================================================
message AckGetFZPswd {
    required uint32  seqno     = 1;      //本包的包序号
    required int32   rslt      = 2;      //-2: 正在登陆; -1: email错误; 0: 服务器发送失败; 1: 成功
}